litmus-portal/graphql-server/pkg/analytics/handler/handler.go

```diff
@@ -966,3 +966,204 @@ func GetScheduledWorkflowStats(projectID string, filter model.TimeFrequency, sho
 
 	return result, nil
 }
+
+func GetWorkflowRunStats(workflowRunStatsRequest model.WorkflowRunStatsRequest) (*model.WorkflowRunStatsResponse, error) {
+	var pipeline mongo.Pipeline
+
+	// Match with projectID
+	matchProjectIdStage := bson.D{
+		{"$match", bson.D{
+			{"project_id", workflowRunStatsRequest.ProjectID},
+		}},
+	}
+	pipeline = append(pipeline, matchProjectIdStage)
+
+	// Match the workflowIds from the input array
+	if len(workflowRunStatsRequest.WorkflowIds) != 0 {
+		matchWfIdStage := bson.D{
+			{"$match", bson.D{
+				{"workflow_id", bson.D{
+					{"$in", workflowRunStatsRequest.WorkflowIds},
+				}},
+			}},
+		}
+
+		pipeline = append(pipeline, matchWfIdStage)
+	}
+
+	includeAllFromWorkflow := bson.D{
+		{"workflow_id", 1},
+		{"workflow_name", 1},
+		{"workflow_manifest", 1},
+		{"cronSyntax", 1},
+		{"workflow_description", 1},
+		{"weightages", 1},
+		{"isCustomWorkflow", 1},
+		{"updated_at", 1},
+		{"created_at", 1},
+		{"project_id", 1},
+		{"cluster_id", 1},
+		{"cluster_name", 1},
+		{"cluster_type", 1},
+		{"isRemoved", 1},
+	}
+
+	// Filter the available workflows where isRemoved is false
+	matchWfRunIsRemovedStage := bson.D{
+		{"$project", append(includeAllFromWorkflow,
+			bson.E{Key: "workflow_runs", Value: bson.D{
+				{"$filter", bson.D{
+					{"input", "$workflow_runs"},
+					{"as", "wfRun"},
+					{"cond", bson.D{
+						{"$eq", bson.A{"$$wfRun.isRemoved", false}},
+					}},
+				}},
+			}},
+		)},
+	}
+	pipeline = append(pipeline, matchWfRunIsRemovedStage)
+
+	// Flatten out the workflow runs
+	unwindStage := bson.D{
+		{"$unwind", bson.D{
+			{"path", "$workflow_runs"},
+		}},
+	}
+	pipeline = append(pipeline, unwindStage)
+
+	// Count all workflowRuns
+	totalWorkflowRuns := bson.A{
+		bson.D{{"$count", "count"}},
+	}
+
+	// Count Succeeded workflowRuns
+	succeededWorkflowRuns := bson.A{
+		bson.D{{"$match", bson.D{
+			{"workflow_runs.phase", model.WorkflowRunStatusSucceeded},
+		}},
+		},
+		bson.D{{"$count", "count"}},
+	}
+
+	// Count Succeeded workflowRuns
+	failedWorkflowRuns := bson.A{
+		bson.D{{"$match", bson.D{
+			{"workflow_runs.phase", model.WorkflowRunStatusFailed},
+		}},
+		},
+		bson.D{{"$count", "count"}},
+	}
+
+	// Count Succeeded workflowRuns
```

Comment:
Can be changed to "// Count Running workflowRuns"

Reply:
Done